### PR TITLE
[DX] Fix metadata cache invalidation. Metadata and tests refactoring

### PR DESCRIPTION
**Pull request by Koc**

### CODE REVIEW

**Comment by garak** [comment]
Is this still WIP or can I merge?

**Comment by Koc** [comment]
@garak it is better merge #624 first, after than I'l do a rebase.

**Comment by garak** [comment]
#624 is merged

**Comment by Koc** [comment]
Seems ok, but I want cover changes with tests (now there is no any tests for xml/yml mapping).

File: Tests/Fixtures/App/src/TestBundle/Entity/Article.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Vich\UploaderBundle\Tests;
+namespace Vich\TestBundle\Entity;
```
Comment (garak) [inline Tests/Fixtures/App/src/TestBundle/Entity/Article.php]: What's "TestBundle"?

**Comment by Koc** [comment]
@garak tests are green. Please review and merge

**Comment by garak** [comment]
Thanks!

**Comment by Koc** [comment]
wow, my refactoring also fixes #704 🤣

**Pull request merged**